server/gokbg3/grails-app/services/org/gokb/TippService.groovy

```diff
@@ -21,18 +27,110 @@ class TippService {
   def reviewRequestService
   def autoTimestampEventListener
 
+  public def updateCoverage(tipp, reqBody) {
+    def cov_list = reqBody.coverageStatements ?: reqBody.coverage
+    def missing = tipp.coverageStatements.collect { it.id }
+    def changed = false
+
+    cov_list?.each { c ->
+      def parsedStart = GOKbTextUtils.completeDateString(c.startDate)
+      def parsedEnd = GOKbTextUtils.completeDateString(c.endDate, false)
+
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'startVolume', c.startVolume)
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'startIssue', c.startIssue)
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'endVolume', c.endVolume)
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'endIssue', c.endIssue)
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'embargo', c.embargo)
+      changed |= com.k_int.ClassUtils.setStringIfDifferent(tipp, 'coverageNote', c.coverageNote)
+      changed |= com.k_int.ClassUtils.setDateIfPresent(parsedStart, tipp, 'startDate')
+      changed |= com.k_int.ClassUtils.setDateIfPresent(parsedEnd, tipp, 'endDate')
```

Comment:
We want to use `updateDateField` instead, since we want to be able remove existing dates as well.

Reply:
done